lib/zone.ts

```diff
@@ -1354,6 +1362,12 @@ const Zone: ZoneType = (function(global: any) {
       }
       error.stack = error.zoneAwareStack = frames.join('\n');
     }
+    // if this is valid, means not undefined
+    // and not global (non-strict mode call Error())
+    // and it is an error not an unrelated object
+    if (this && this !== global && this instanceof Error) {
+      error = setPrototypeOf(error, getPrototypeOf(this));
```

Comment:
Calling `setProtypeOf` will make this object slow. Not sure if that is a problem since it is a Error object.